bundle/Resources/public/js/views/plugins/cof-selectcontenttypeplugin.js

```diff
@@ -16,6 +16,70 @@ YUI.add('cof-selectcontenttypeplugin', function (Y) {
     Y.cof.Plugin.SelectContentType = Y.Base.create('selectContentType', Y.eZ.Plugin.ContentCreate, [], {
         initializer: function () {
             this.onHostEvent('*:fetchContentTypes', this._getContentTypes, this);
+            this.onHostEvent('*:fetchSuggestedLocations', this._getSuggestedLocations, this);
+        },
+
+        /**
+         * Get the suggested locations
+         *
+         * @protected
+         * @method _getSuggestedLocations
+         * @param event {Object} event facade
+         */
+        _getSuggestedLocations: function (event) {
+            var contentTypeId = '/' + event.newVal.get('identifier');
+
+            Y.io(this.get('host').get('app').get('apiRoot') + 'api/ezp/v2/contentonthefly/locations' + contentTypeId, {
+                on: {
+                    success: Y.bind(function (id, xhr) {
+                        var locationSchema = {resultListLocator: 'Location', resultFields: [{key:'href', locator: '@href'}]},
+                            locations = Y.DataSchema.XML.apply(locationSchema, xhr.responseXML);
+
+                        this._loadLocations(locations.results)
+                            .then(function (result) {
+                                event.target.set('suggestedLocations', result);
+                            });
+                    }, this)
+                }
+            });
+        },
+
+        /**
+         * Loads suggested locations
+         *
+         * @protected
+         * @method _loadLocations
+         * @param locationsList {Array} the suggested locations list
+         */
+        _loadLocations: function (locationsList) {
+            var capi = this.get('host').get('capi'),
+                contentService = this.get('host').get('capi').getContentService(),
+                promises = locationsList.map(function (location) {
+                    return new Y.Promise(function (resolve, reject) {
+                        contentService.loadLocation(location.href, function (error, response) {
+                            var locationModel;
+
+                            if (error) {
+                                reject(error, response);
+
+                                return;
+                            }
+
+                            locationModel = new Y.eZ.Location({
+                                id: response.document.Location._href,
+                                locationId: response.document.Location.id,
+                                pathString: response.document.Location.pathString,
+                                contentInfo: response.document.Location.ContentInfo
+                            });
+
+                            locationModel.loadPath({api: capi}, function () {
```

Comment:
Nitpick: you can do something like `locationModel.loadPath({api: capi}, Y.bind(resolve, this, locationModel))`